common/src/main/java/com/wynntils/mc/mixin/GuiMixin.java

```diff
@@ -56,4 +60,46 @@ public void onRenderGuiPre(PoseStack poseStack, CallbackInfo ci) {
             ci.cancel();
         }
     }
+
+    @Inject(method = "renderHearts", at = @At(value = "HEAD"), cancellable = true)
+    public void onRenderHeartsPre(
+            PoseStack poseStack,
+            Player player,
+            int x,
+            int y,
+            int height,
+            int i,
+            float f,
+            int j,
+            int k,
+            int l,
+            boolean bl,
+            CallbackInfo ci) {
+        if (EventFactory.onRenderHearthsPre(poseStack, this.minecraft.getWindow())
+                .isCanceled()) {
+            ci.cancel();
+        }
+
+        RenderSystem.setShaderTexture(0, GuiComponent.GUI_ICONS_LOCATION); // we have to reset shader texture
```

Comment:
Why is that? Is this needed even if no-one listens to the event? Or is the event handler messing things up? If so, should not they clear this up?

Reply:
I think it is to set the state to the original one in the mixin, instead of the things that use this event.